airbyte-commons/src/test/java/io/airbyte/commons/json/JsonPathsTest.java

```diff
@@ -0,0 +1,228 @@
+    assertTrue(JsonPaths.isPathPresent(JSON_NODE, NESTED_FIELD_QUERY));
+    assertTrue(JsonPaths.isPathPresent(JSON_NODE, JSON_OBJECT_QUERY));
+    assertFalse(JsonPaths.isPathPresent(JSON_NODE, EMPTY_RETURN_QUERY));
+  }
+
+  @Test
+  void testReplaceAtStringLoud() {
+    assertOriginalObjectNotModified(JSON_NODE, () -> {
+      final JsonNode expected = Jsons.clone(JSON_NODE);
+      ((ArrayNode) expected.get("one")).set(1, REPLACEMENT_STRING);
+
+      final JsonNode actual = JsonPaths.replaceAtStringLoud(JSON_NODE, LIST_ONE_QUERY, REPLACEMENT_STRING);
+      assertEquals(expected, actual);
+    });
+  }
+
+  @SuppressWarnings("CodeBlock2Expr")
+  @Test
+  void testReplaceAtStringLoudEmptyPathThrows() {
+    assertOriginalObjectNotModified(JSON_NODE, () -> {
+      assertThrows(PathNotFoundException.class, () -> JsonPaths.replaceAtStringLoud(JSON_NODE, EMPTY_RETURN_QUERY, REPLACEMENT_STRING));
+    });
+  }
+
+  @Test
+  void testReplaceAtString() {
+    assertOriginalObjectNotModified(JSON_NODE, () -> {
+      final JsonNode expected = Jsons.clone(JSON_NODE);
+      ((ArrayNode) expected.get("one")).set(1, REPLACEMENT_STRING);
+
+      final JsonNode actual = JsonPaths.replaceAtString(JSON_NODE, LIST_ONE_QUERY, REPLACEMENT_STRING);
+      assertEquals(expected, actual);
+    });
+  }
+
+  @Test
+  void testReplaceAtStringEmptyReturnNoOp() {
+    assertOriginalObjectNotModified(JSON_NODE, () -> {
+      final JsonNode expected = Jsons.clone(JSON_NODE);
+
+      final JsonNode actual = JsonPaths.replaceAtString(JSON_NODE, EMPTY_RETURN_QUERY, REPLACEMENT_STRING);
+      assertEquals(expected, actual);
+    });
+  }
+
+  @Test
+  void testReplaceAtJsonNodeLoud() {
+    assertOriginalObjectNotModified(JSON_NODE, () -> {
+      final JsonNode expected = Jsons.clone(JSON_NODE);
+      ((ArrayNode) expected.get("one")).set(1, REPLACEMENT_JSON);
+
+      final JsonNode actual = JsonPaths.replaceAtJsonNodeLoud(JSON_NODE, LIST_ONE_QUERY, REPLACEMENT_JSON);
+      assertEquals(expected, actual);
+    });
+  }
+
+  @SuppressWarnings("CodeBlock2Expr")
+  @Test
+  void testReplaceAtJsonNodeLoudEmptyPathThrows() {
+    assertOriginalObjectNotModified(JSON_NODE, () -> {
+      assertThrows(PathNotFoundException.class, () -> JsonPaths.replaceAtJsonNodeLoud(JSON_NODE, EMPTY_RETURN_QUERY, REPLACEMENT_JSON));
+    });
+  }
+
+  @Test
+  void testReplaceAtJsonNodeLoudMultipleReplace() {
+    assertOriginalObjectNotModified(JSON_NODE, () -> {
+      final JsonNode expected = Jsons.clone(JSON_NODE);
+      ((ArrayNode) expected.get("one")).set(0, REPLACEMENT_JSON);
+      ((ArrayNode) expected.get("one")).set(1, REPLACEMENT_JSON);
+      ((ArrayNode) expected.get("one")).set(2, REPLACEMENT_JSON);
+
+      final JsonNode actual = JsonPaths.replaceAtJsonNodeLoud(JSON_NODE, LIST_ALL_QUERY, REPLACEMENT_JSON);
+      assertEquals(expected, actual);
+    });
+  }
+
+  // todo (cgardens) - this behavior is a little unintuitive, but based on the docs, there's not an
```

Comment:
not sure how intuitive this is to others. was a bit weird for me but inclined to leave it as be.